awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -403,6 +404,12 @@ def authenticate(self, **kwargs):
                     force_basic_auth=True, url_username=self.username, url_password=self.password,
                     data=dumps(login_data), headers={'Content-Type': 'application/json'}
                 )
+            except HTTPError as he:
+                try:
+                    resp = he.read()
+                except Exception as e:
+                    resp = 'unknown {0}'.format(e)
+                self.fail_json(msg='Failed to get token: {0} response: {1}'.format(he, resp))
```

Comment:
Do you want to put the response directly in this message or just add it to the return hash and reference it? i.e. 
```
self.fail_json(**{msg='Failed to get token: {0}'.format(he), response=resp})
```

Reply:
okay that is actually really interesting. So people get it when they run it verbose. I might do that.

Reply:
1 of the 2 cases is a failure, and the other is a warning, so I had to munge the text together for the warning still.